lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -254,11 +310,118 @@ describe('BasicLtiLaunchApp', () => {
     });
   });
 
+  describe('concurrent fetching of groups and content', () => {
+    let contentUrlCall;
+    let groupsCall;
+
+    beforeEach(() => {
+      // Will attempt to fetch the 1. content url and 2. groups.
+      fakeConfig.api = {
+        authToken: 'dummyAuthToken',
+        viaCallbackUrl: 'https://lms.hypothes.is/api/files/1234',
+        sync: {
+          data: {
+            course: {
+              context_id: '12345',
+              custom_canvas_course_id: '101',
+            },
+          },
+          path: '/api/sync',
+        },
+      };
+      contentUrlCall = fakeApiCall.withArgs({
+        authToken: 'dummyAuthToken',
+        path: 'https://lms.hypothes.is/api/files/1234',
+      });
+
+      groupsCall = fakeApiCall.withArgs({
+        authToken: 'dummyAuthToken',
+        path: '/api/sync',
+        data: {
+          course: {
+            context_id: '12345',
+            custom_canvas_course_id: '101',
+          },
+        },
+      });
+    });
+
+    it.skip('renders the spinner until both groups and contentUrl requests finish', async () => {
```

Comment:
This test was a false negative and was in fact not correct before.

I changed it to use `waitForElementToRemove` and that needs to fail. I would in fact use `assert.rejects` but that has not been ported to LMS yet. But in any case, this test fails.

It seems that the groups and contentUrl just resolve right away without needing to call `await`. That means I can't pause on an in-between state between `groups` and `contentUrl` to check that the spinner is still spinning.

`waitForElementToRemove` is a bit more accurate than just calling update() once, and in this case it does prove to be different. What I really want to show is: if `groups` does not reject or resolve, then still show a spinner.